cifopy/charles/charles.py

```diff
@@ -108,9 +124,34 @@ def evolve(self, gens, select, crossover, mutate, co_p, mu_p, elitism):
             self.individuals = new_pop
  
             if self.optim == "max":
-                print(f'Best Individual: {max(self, key=attrgetter("fitness"))}')
+                if gen%50==0:
+                    print(f'Best Individual: {max(self, key=attrgetter("fitness"))} Gen: {gen}')
+ 
             elif self.optim == "min":
-                print(f'Best Individual: {min(self, key=attrgetter("fitness"))}')
+                if gen%50==0:
+                    print(f'Best Individual: {min(self, key=attrgetter("fitness"))} Gen: {gen}')
+    
+    def reshuffle(self):
+        n = len(self)
+        new_pop = []
+        if self.optim == "max":
+            new_pop.append(deepcopy(max(self.individuals, key=attrgetter("fitness"))))
+        elif self.optim == "min":
+            new_pop.append(deepcopy(min(self.individuals, key=attrgetter("fitness"))))
```

Comment:
This is to keep the best individual, right? If elitism is used, I think we do it twice. Do you think it's a problem?

Reply:
Elitism comes to a different stage, if we didn't do this we would potentially lose the best. A different issue in the immediate following lines where we save half of the current population. In this way we could have the best individual twice (50% of the times, there should be a comment in the code). I don't think this is a big issue but we maybe is best to change it in order to be more consistent.